### PR TITLE
Add `-index-include-locals` flag.

### PR DESCRIPTION
**Pull request by allevato**
When this flag is passed, index store data emitted during a build
will also contain definitions/references for local symbols.

### CODE REVIEW

**Comment by allevato** [comment]
@swift-ci please test

File: test/Index/Store/record-include-locals.swift [hunk 1]
```diff
@@ -0,0 +1,88 @@
+// RUN: rm -rf %t
```
Comment (bnbarham) [inline test/Index/Store/record-include-locals.swift:1]: I'd probably be fine with just `driver-include-locals.swift` + the existing test in `test/Index/local.swift`. Could print jobs in that one as well and verify `-index-locals` is in the frontend command.

**Review by bnbarham — COMMENTED** [review]
This will also need a corresponding change in `swift-driver`.

**Comment by bnbarham** [comment]
@swift-ci please test

File: test/Index/Store/record-include-locals.swift (hunk 1 above)
Comment (allevato) [reply test/Index/Store/record-include-locals.swift]: Done (added checks to print the jobs, and kept the small example to verify that the frontend indeed emits the locals).

**Review by bnbarham — APPROVED** [review]
Nice PR number by the way, I'm surprised no one made a dummy PR to get it 😆!

**Comment by allevato** [comment]
> This will also need a corresponding change in `swift-driver`.

Drafted in https://github.com/apple/swift-driver/pull/1129

**Comment by bnbarham** [comment]
@swift-ci please test

**Pull request merged**